src/core_modules/capture-core/components/DataEntry/withSaveHandler/withSaveHandler.js

```diff
@@ -184,18 +185,17 @@ const getSaveHandler = (
         }
 
         handleSaveDialog = () => {
-            this.handleSave(this.state.saveType);
+            this.handleSave(addEventSaveTypes.SAVE_WITHOUT_COMPLETING);
             this.setState({ messagesDialogOpen: false });
         }
 
         handleSave = (saveType?: ?string) => {
-            const { onSave, itemId, id, calculatedFoundation } = this.props;
-            onSave(
-                itemId,
-                id,
-                calculatedFoundation,
-                saveType,
-            );
+            const { onSave, itemId, id, calculatedFoundation, warnings, errors } = this.props;
+            if (saveType === addEventSaveTypes.COMPLETE && ((errors && errors.length > 0) || (warnings && warnings.length > 0))) {
+                this.showMessagesPopup(saveType);
```

Comment:
hi @simonadomnisoru  Why do we only show the message when the save type is COMPLETE?
should it show if there is error and warning even when save type is SAVE_WITHOUT_COMPLETING?

Reply:
Hey, 
If the user clicks the button "Save without completing", we allowed for the event to be saved without validating the fields. The error popup will be shown only when the user clicks the "Complete" button. [DHIS2-11669](https://jira.dhis2.org/browse/DHIS2-11669) 
Thanks!